solutions/silver/cf-1744F.mdx

```diff
@@ -0,0 +1,144 @@
+---
+id: cf-1744F
+source: CF
+title: MEX vs MED
+author: Alex Du
+---
+
+## Explanation
+
+[Official Analysis (No Code)](https://codeforces.com/blog/entry/108101)
+
+<Spoiler title="Hint 1">
+
+How do we know if a specific subsegment satisfies the constraints for a given
+$\texttt{mex}$?
+
+First, to reach that value of $\texttt{mex}$, we must find subsegments which contain all
+integers $0, 1, \ldots, \texttt{mex}-1$, while also not containing $\texttt{mex}$.
+
+Then, given a subsegment, $\texttt{mex} > \texttt{med}$ will be satisfied if the length of the
+subsegment is $\leq 2 \cdot \texttt{mex}$, since the median will be at most $\texttt{mex}-1$.
+
+</Spoiler>
+
+<Spoiler title="Hint 2">
+
+Combining the two ideas in Hint 1, we can first find $l$ and $r$, which represent the
+indices of the left-most and right-most occurences of the numbers
+$0, 1, \ldots \texttt{mex} - 1$ respectively. Then, find all subsegments that contain the
+subsegment $l \ldots r$ within themselves, have length $\leq 2 \cdot \texttt{mex}$, and
+do not contain $\texttt{mex}$ itself. Now, we just need to count these
+subsegments efficiently.
```

Comment:
ngl i feel like this is too specific
@ryanchou-dev do you agree with me that hints should give a general idea and not go into specifics w/ variables n stuff?

Reply:
left it out and put in solution